tools/scripts/xabuild

```diff
@@ -133,7 +133,11 @@ case "$MSBUILD" in
 		XABUILD_FLAGS+=(/p:TargetFrameworkRootPath="$TARGETS_DIR-frameworks/")
 		;;
 	*xbuild*)
-		export XBUILD_FRAMEWORK_FOLDERS_PATH="$TARGETS_DIR-frameworks"
+		if [ -z "$XBUILD_FRAMEWORK_FOLDERS_PATH" ]; then
+			export XBUILD_FRAMEWORK_FOLDERS_PATH="$TARGETS_DIR-frameworks"
+		else 
+			export XBUILD_FRAMEWORK_FOLDERS_PATH="$XBUILD_FRAMEWORK_FOLDERS_PATH"
```

Comment:
Why does the environment have `$XBUILD_FRAMEWORK_FOLDERS_PATH` already exported, and why should we be using it *in preference to the computed value*?

Reply:
We need to be able to override it to the Faux location as part of the unit tests. If we don't it will pick up the `bin\Debug` path which we don't want. Is there a better way of doing it?

Reply:
No, I can't think of a better way, though this will only work with `xbuild`, not `msbuild`.

Perhaps a "better" (or just different?) way would be to define a *new* environment variable, e.g. `$__MSBUILD_FRAMEWORK_FOLDERS_PATH_OVERRIDE`, and use that if it's set. We'd also want to update the `msbuild`-based codepath to follow suit, perhaps via [setting `$(TargetFrameworkRootPath)`](https://github.com/xamarin/xamarin-android/blob/49ecddaea61d789486d5d7b9724d842d495f1257/tools/scripts/xabuild#L133)?

One problem with the current `xabuild` script, though: the `msbuild`-specific logic on line 133 happens *after* we [`exec xabuild.exe` on line 86](https://github.com/xamarin/xamarin-android/blob/49ecddaea61d789486d5d7b9724d842d495f1257/tools/scripts/xabuild#L86), meaning line 133/etc. is superfluous. This should be corrected.

Reply:
msbuild will use the `TargetFrameworkRootPath` as a parameter so msbuild support works. xbuild ignores or doesn't use `TargetFrameworkRootPath` so thats why I needed to use the envvar.